src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -390,9 +390,13 @@ export abstract class AbstractSearchAndReplaceAction extends Action {
 	/**
 	 * Returns element to focus after removing the given element
 	 */
-	getElementToFocusAfterRemoved(viewer: WorkbenchObjectTree<RenderableMatch>, elementToBeRemoved: RenderableMatch): RenderableMatch {
-		const elementToFocus = this.getNextElementAfterRemoved(viewer, elementToBeRemoved);
-		return elementToFocus || this.getPreviousElementAfterRemoved(viewer, elementToBeRemoved);
+	getElementToFocusAfterRemoved(viewer: WorkbenchObjectTree<RenderableMatch>, lastElementToBeRemoved: RenderableMatch, firstElementToBeRemoved?: RenderableMatch): RenderableMatch {
+		if (!firstElementToBeRemoved) {
+			firstElementToBeRemoved = lastElementToBeRemoved;
+		}
+
+		const elementToFocus = this.getNextElementAfterRemoved(viewer, lastElementToBeRemoved);
```

Comment:
Looking at the first and last elements doesn't seem like enough. If elements A, B, D, and E are deleted, shouldn't C be the one that gets selected? Will that work here?